crates/ruff_testing_macros/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "ruff_testing_macros"
```

Comment:
In my view it mainly comes down if you prefer having fewer crates or more fine granular crates. 

My preference is to have more fine granular crates because:

* Smaller crates mean there are fewer changes that require a re-compile of its code
* You can be explicit about dependencies
  * the testing macros don't need to depend on itertools
  * The test macros can be a `dev` dependency, ensuring that its functionality isn't used in production code
* I find smaller crates help navigate the code base and specific names tell you if this crate is the right crate for your code addition

The main downside is that we end up with more crates, resulting in a longer `crates` list. This doesn't seem to be a problem today.